frontend/src/metabase/visualizations/visualizations/CartesianChart/use-chart-events.ts

```diff
@@ -86,6 +87,16 @@ export const useChartEvents = (
             return;
           }
 
+          if (
+            event.name === "goal-line" &&
+            event.event?.target?.style?.text === settings["graph.goal_label"]
```

Comment:
That make sense, I think you can change the check to `target.nodeName === "text"`